a10_neutron_lbaas/a10_openstack_lb.py

```diff
@@ -33,9 +33,12 @@
 
 class A10OpenstackLBBase(object):
 
-    def __init__(self, openstack_driver, plumbing_hooks_class=hooks.PlumbingHooks):
+    def __init__(self, openstack_driver,
+                 plumbing_hooks_class=hooks.PlumbingHooks,
+                 neutron_hooks_module=None):
```

Comment:
I like the dependency injection in the constructor.  I prefer constructor-based DI because it prevents us from creating the class without satisfying dependencies and is more of a static typing approach.

Validating that the dependency implements the expected class would be my only suggestion.